tensorboard/webapp/widgets/histogram/histogram_fob_adapter_test.ts

```diff
@@ -0,0 +1,158 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {Component} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {ScaleLinear} from '../../third_party/d3';
+import {HistogramFobAdapter} from './histogram_fob_adapter';
+
+@Component({
+  selector: 'axis-overlay',
+  template: ` <div></div> `,
+})
+class AxisOverlayComponent {}
+
+describe('HistogramFobAdapter', () => {
+  let temporalScaleSpy: jasmine.Spy;
+  beforeEach;
+  function createInterface(customSteps?: number[]) {
+    temporalScaleSpy = jasmine.createSpy();
+    let scale = temporalScaleSpy as unknown as ScaleLinear<number, number>;
+    let steps = customSteps || [0, 1, 2, 3];
+    temporalScaleSpy.and.callFake((step: number) => {
+      return step;
+    });
+    return new HistogramFobAdapter(scale, steps);
+  }
+  describe('setBounds', () => {
```

Comment:
These types of tests are ok -- in a lot of cases testing the interface is best practice. But I think it is important to also test the combination of LinkedTimeFobControllerComponent + HistogramFobAdapter since there are some important interactions there.  So you could either:

1) Supplement these tests with webtests (If you go this route I'd suggest trying to write those webtests side x side with this PR so you can get a sense if it will actually work)
2) Supplement these tests with unit tests that combine LinkedTimeFobControllerComponent + HistogramFobAdapter.

I suspect option (2) is easier for histograms only because these tests probably already exist in some form.

If you do option (2) you might find it a good idea to create a first-class histogram_fob_controller_component that does all the wiring for you.